py/selenium/webdriver/remote/webelement.py

```diff
@@ -91,11 +90,20 @@ def click(self) -> None:
 
     def submit(self):
         """Submits a form."""
-        form = self.find_element(By.XPATH, "./ancestor-or-self::form")
-        self._parent.execute_script(
-            "var e = arguments[0].ownerDocument.createEvent('Event');"
-            "e.initEvent('submit', true, true);"
-            "if (arguments[0].dispatchEvent(e)) { arguments[0].submit() }", form)
+        script = "var form = arguments[0];\n" \
+                 "while (form.nodeName != \"FORM\" && form.parentNode) {\n" \
+                 "  form = form.parentNode;\n" \
+                 "}\n" \
+                 "if (!form) { throw Error('Unable to find containing form element'); }\n" \
+                 "if (!form.ownerDocument) { throw Error('Unable to find owning document'); }\n" \
+                 "var e = form.ownerDocument.createEvent('Event');\n" \
+                 "e.initEvent('submit', true, true);\n" \
+                 "if (form.dispatchEvent(e)) { HTMLFormElement.prototype.submit.call(form) }\n"
+
+        try:
+            self._parent.execute_script(script, self)
+        except JavascriptException:
```

Comment:
Not a blocker but this is a change in exception thrown and we should be aware of that

Reply:
We could continue to throw the JS error and just update the message if you would prefer that. JS error just seemed confusing/non-obvious to me.

Reply:
I think it's ideal to keep the same as before (given it's not a result of a completely broken functionality?) Could just raise the same with a new message ` raise JavascriptException exception(msg) from None` (assuming it raises that previously)

Reply:
This is what I get for trying to reply via my phone. I thought we were talking Java bindings.

The current Python code raises an "Element Not Found" error for this scenario. Updating to the all-JS approach that Java is currently using gives an unhelpful JavaScript error message. This line isn't what is changing existing behavior, it is just tailoring the new behavior.